src/guidellm/backend/deepsparse/backend.py

```diff
@@ -15,11 +17,30 @@ class DeepsparseBackend(Backend):
     """
 
     def __init__(self, model: Optional[str] = None, **request_args):
-        self.request_args = request_args
-        self.pipeline: Pipeline = Pipeline.create(
-            task="sentiment-analysis",
-            model_path=model or self.default_model,
-        )
+        self._request_args: Dict[str, Any] = request_args
+        self.model: str = self._get_model(model)
+        self.pipeline: Pipeline = TextGeneration(model=self.model)
+
+    def _get_model(self, model_from_cli: Optional[str] = None) -> str:
+        """Provides the model by the next priority list:
+        1. from function argument (comes from CLI)
+        1. from environment variable
+        2. `self.default_model` from `self.available_models`
+        """
+
+        if model_from_cli is not None:
+            return model_from_cli
+        elif settings.deepsprase.model is not None:
```

Comment:
deepsprase should be deepsparse in a few places

Reply:
Oh, nice catch. Thanks!